vllm/attention/layer.py

```diff
@@ -360,12 +360,9 @@ def __init__(
             # currently, only torch_sdpa is supported on rocm
             self.attn_backend = _Backend.TORCH_SDPA
         else:
-            if backend in (_Backend.FLASH_ATTN, _Backend.FLASH_ATTN_VLLM_V1,
-                           _Backend.FLEX_ATTENTION):
-                backend = _Backend.XFORMERS
-
             self.attn_backend = backend if backend in {
-                _Backend.TORCH_SDPA, _Backend.XFORMERS, _Backend.PALLAS_VLLM_V1
+                _Backend.TORCH_SDPA, _Backend.XFORMERS, _Backend.PALLAS_VLLM_V1,
+                _Backend.ROCM_AITER_FA, _Backend.FLEX_ATTENTION
             } else _Backend.TORCH_SDPA
```

Comment:
```suggestion
                _Backend.TORCH_SDPA, _Backend.XFORMERS, _Backend.PALLAS_VLLM_V1,
                _Backend.ROCM_AITER_FA,
            } else current_platform.get_vit_attn_backend()
```
We can remove FlexAttention here to use default fallback.